oauth2client/contrib/appengine.py

```diff
@@ -89,11 +89,11 @@ def _safe_html(s):
     return cgi.escape(s, quote=1).replace("'", '&#39;')
 
 
-class InvalidClientSecretsError(Exception):
-    """The client_secrets.json file is malformed or missing required fields."""
+class Error(Exception):
+    """Base error for this module."""
 
 
-class InvalidXsrfTokenError(Exception):
+class InvalidXsrfTokenError(Error):
```

Comment:
I think we can just catch the exception within this module and send the appropriate error response.
https://github.com/google/oauth2client/blob/master/oauth2client/contrib/appengine.py#L816-817

```python
redirect_uri = _parse_state_value(str(self.request.get('state')), user)
```
change to:
```python
try:
    redirect_uri = _parse_state_value(str(self.request.get('state')), user)
except InvalidXsrfTokenError:
    self.response.out.write('The authorization request failed: Invalid CSRF Token')
    return # or redo some other logic, so that self.redirect isn't executed below
```

or something along those lines.